packages/go/lab/loader.go

```diff
@@ -0,0 +1,110 @@
+	// `object_id` is the main identifier we use in BloodHound.
+	Properties map[string]any `json:"properties"`
+}
+
+// Edge is the JSON representation of a graph edge.
+type Edge struct {
+	// FromID is the local Node identifier for the start of an Edge.
+	FromID string `json:"fromId"`
+
+	// ToID is the local Node identifier for the end of an Edge.
+	ToID string `json:"toId"`
+
+	// Type is the 'label' we apply to the Edge. This is synonymous to
+	// the edge Kind in BloodHound.
+	Type string `json:"type"`
+
+	// Properties is the key:value map used for storing extra information
+	// on the Edge.
+	Properties map[string]any `json:"properties"`
+}
+
+// ParseGraphFixtureJsonFile takes in a fs.File interface and parses its contents into a
+// GraphFixture struct.
+func ParseGraphFixtureJsonFile(fh fs.File) (GraphFixture, error) {
+	var graphFixture GraphFixture
+	if err := json.NewDecoder(fh).Decode(&graphFixture); err != nil {
+		return graphFixture, fmt.Errorf("could not unmarshal graph data: %w", err)
+	} else {
+		return graphFixture, nil
+	}
+}
+
+// LoadGraphFixture requires a graph.Database interface and a GraphFixture struct.
+// It will import the nodes and edges from the GraphFixture and inserts them into
+// the graph database. It uses the `id` property of the nodes as the local
+// identifier and then maps them to database IDs, meaning that the `id` given in
+// the file will not be preserved.
+func LoadGraphFixture(db graph.Database, g *GraphFixture) error {
+	var nodeMap = make(map[string]graph.ID)
+	if err := db.WriteTransaction(context.Background(), func(tx graph.Transaction) error {
+		for _, node := range g.Nodes {
+			if dbNode, err := tx.CreateNode(graph.AsProperties(node.Properties), graph.StringsToKinds(node.Labels)...); err != nil {
+				return fmt.Errorf("could not create node `%s`: %w", node.ID, err)
+			} else {
+				nodeMap[node.ID] = dbNode.ID
+			}
+		}
+		for _, edge := range g.Relationships {
+			if startId, ok := nodeMap[edge.FromID]; !ok {
+				return fmt.Errorf("could not find start node %s", edge.FromID)
+			} else if endId, ok := nodeMap[edge.ToID]; !ok {
+				return fmt.Errorf("could not find end node %s", edge.ToID)
+			} else if _, err := tx.CreateRelationshipByIDs(startId, endId, graph.StringKind(edge.Type), graph.AsProperties(edge.Properties)); err != nil {
+				return fmt.Errorf("could not create relationship `%s` from `%s` to `%s`: %w", edge.Type, edge.FromID, edge.ToID, err)
+			}
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("error writing graph data: %w", err)
+	}
+	return nil
+}
+
+// LoadGraphFixtureFile takes a graph.Database interface, a fs.FS interface,
+// and a path string. It will attempt to read the given path from the FS and
+// parse then file into a GraphFixture, and finally import the GraphFixture
+// Nodes and Edges into the graph.Database.
+func LoadGraphFixtureFile(db graph.Database, fSys fs.FS, path string) error {
+	if fh, err := fSys.Open(path); err != nil {
```

Comment:
Let's avoid using `if err :=` when it becomes nested. I'd rather see a more traditional `if err != nil` for the outer in this case

Reply:
Not sure I understand the reasoning behind this one. If it's to avoid shadowing the `err`, it still gets shadowed even if the outer `if` is collapsed

Reply:
Nope, it's because nested chaining is awful and I seek to remove it. We can take this offline if you'd like.